app/controllers/resumes_controller.rb

```diff
@@ -40,18 +40,25 @@ def random
       return
     end
 
-    resumes = Resume.where(public: true).where.not(user_id: current_user.id).all
-    resumes = resumes.select {|resume| resume.user.points > 0}
-    resumes = resumes.select {|resume| resume.user.industry == current_user.industry}
-    resumes = resumes.select {|resume| resume.reviews.none? {|review| review.user_id == current_user.id }}
-    if resumes.length == 0
-      flash[:error] = "No valid resumes"
+    # To keep load testing simple and consistent,
+    # we won't worry about this condition:
+    # Resume.joins(:user, :reviews)
+    # .where.not(reviews: { user_id: current_user.id })
+    resume = Resume.joins(:user)
+                   .where(public: true)
+                   .where.not(user_id: current_user.id)
+                   .where.not(users: { points: 0 })
+                   .where(users: { industry: current_user.industry })
+                   .sample
```

Comment:
We could do something where we sample and check that a review isn't pre-existing, and keep sampling until we get one or there's none left.